src/components/Main/Main.js

```diff
@@ -1,17 +1,32 @@
 import Carousel from "../Carousel/Carousel";
 import Slider from "../Slider/Slider";
-import productCategoriesMock from "../../mocks/en-us/product-categories.json";
-import featuredBannersMock from "../../mocks/en-us/featured-banners.json";
-import featuredProductsMock from "../../mocks/en-us/featured-products.json";
 import FeaturedProducts from "../FeaturedProducts/FeaturedProducts";
+import { useFeaturedBanners } from "../../utils/hooks/useFeaturedBanners";
+import { useFeaturedProducts } from "../../utils/hooks/useFeaturedProducts";
+import ShareDataContext from "../../context/shareDataContext";
+import { useContext } from "react";
 
 const Main = () => {
+  const featuredBanners = useFeaturedBanners();
+  const featuredProducts = useFeaturedProducts();
+  const productCategories = useContext(ShareDataContext).productCategories;
 
   return (
     <main>
-        <Slider items={featuredBannersMock.results} />
-        <Carousel items={productCategoriesMock.results} />
-        <FeaturedProducts items={featuredProductsMock.results} />
+      {featuredBanners.isLoading ||
+      productCategories.isLoading ||
+      featuredProducts.isLoading ? (
```

Comment:
As a standard is better move the conditions outside of the `return` part, and use `constants` with descriptive names, the reason is for readability for other people.

```suggestion
const showLoading = featuredBanners.isLoading ||
      productCategories.isLoading ||
      featuredProducts.isLoading;
      
     //...
     
      {showLoading ? (
```

Reply:
done